feat(STONEINTG-979): add gitlab and environment tests in README (#1236)
konflux-ci/e2e-tests: tests/integration-service/README.md

### E2E tests of Status Reporting for Integration tests PLR in CommitStatus over GitLab.com (within `gitlab-integration-reporting.go`):

- Creating 2 IntegrationTestScenarios: one that's supposed to pass and other one to fail.
- Creating PaC branch and component base branches which will be used to create Merge Request (MR).
- Testing for successful creation of applications and component (with the above custom branch).
- Checking if the Build pipelineRun is successfully triggered and completed.
- Asserting the creation of PaC MR within the component repo.
- Asserting the proper status reporting of Build PipelineRun within the MR's CommitStatus.
- Asserting the signing of BuildPipelineRun.
- Asserting the successful creation of a Snapshot after a push event.
- Verifying that both the Integration pipelineRuns got created and finished successfully.
- Checking that a Snapshot gets successfully marked as 'failed' after all Integration pipelinRuns finished.
- Asserting the proper status reporting of both the Integration pipelineRuns within the MR's CommitStatus.

Happy path testing describes tests while integration test scenario pipelineRun reuns over environment.

> ### E2E Tests with Environrmnrt (within `integration-with-env.go`):

Reviewer: ```suggestion
### E2E Tests with Environment (within `integration-with-env.go`):
```

- Testing for successful creation of integration test scenario pointed to pipeline: integration-examples/pipelines
/integration_resolver_pipeline_environment_pass.yaml.
- Testing for successful creation of applications and components.
- Checking if the BuildPipelineRun is successfully triggered, contains the finalizer, and got completed.
- Asserting the signing of BuildPipelineRun.
- Validating the successful creation of a Snapshot, and removal of finalizer from BuildPipelineRun.
- Verifying that CronJob "spacerequest-cleaner" exists.
- Verifying that the Integration PipelineRuns finished successfully.
- Checking that the status of integration test was reported to the Snapshot.

> - Checking that a snapshot gets successfully marked as 'passed' after Integration pipeline runs finished.

Reviewer: nit:
```suggestion
- Checking that a snapshot gets successfully marked as 'passed' after Integration PipelineRuns finished.
```

- Checking if space request is created in namespace.

> - Verify that space request is deleted after successful deletion of integration test scenario pipelinerun.

Reviewer: ```suggestion
- Verify that space request is deleted after successful deletion of Integration PipelineRun.
```

- Asserting the proper status reporting of both the Integration pipelineRuns within the PR's CheckRuns.

### E2E tests of Status Reporting for Integration tests PLR in CommitStatus over GitLab.com (within `gitlab-integration-reporting.go`):

- Creating 2 IntegrationTestScenarios: one that's supposed to pass and other one to fail.
- Creating PaC branch and component base branches which will be used to create Merge Request (MR).
- Testing for successful creation of applications and component (with the above custom branch).
- Checking if the Build pipelineRun is successfully triggered and completed.
- Asserting the creation of PaC MR within the component repo.
- Asserting the proper status reporting of Build PipelineRun within the MR's CommitStatus.
- Asserting the signing of BuildPipelineRun.
- Asserting the successful creation of a Snapshot after a push event.
- Verifying that both the Integration pipelineRuns got created and finished successfully.
- Checking that a Snapshot gets successfully marked as 'failed' after all Integration pipelinRuns finished.
- Asserting the proper status reporting of both the Integration pipelineRuns within the MR's CommitStatus.

> Happy path testing describes tests while integration test scenario pipelineRun reuns over environment.

Author: this irrelevant line, 
will remove

### E2E Tests with Environrmnrt (within `integration-with-env.go`):

- Testing for successful creation of integration test scenario pointed to pipeline: integration-examples/pipelines
/integration_resolver_pipeline_environment_pass.yaml.
- Testing for successful creation of applications and components.
- Checking if the BuildPipelineRun is successfully triggered, contains the finalizer, and got completed.
- Asserting the signing of BuildPipelineRun.
- Validating the successful creation of a Snapshot, and removal of finalizer from BuildPipelineRun.
- Verifying that CronJob "spacerequest-cleaner" exists.
- Verifying that the Integration PipelineRuns finished successfully.
- Checking that the status of integration test was reported to the Snapshot.
- Checking that a snapshot gets successfully marked as 'passed' after Integration pipeline runs finished.
- Checking if space request is created in namespace.

> - Verify that after deleting successfully of integration test scenario pipelineRun, the space request also deleted.

Reviewer 2: ```suggestion
- Verify that space request is deleted after successful deletion of integration test scenario pipelinerun.
```

Author: agree, 
your suggestion reads better, 
fixed
